ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -173,22 +173,38 @@ function groupNodes(nodes: any[], graph: dagre.graphlib.Graph) {
         groupedNodesArr.forEach((obj: {kind: string; nodeIds: string[]; parentIds: dagre.Node[]}) => {
             const {nodeIds, kind, parentIds} = obj;
             const groupedNodeIds: string[] = [];
+            const podGroupIds: string[] = [];
             nodeIds.forEach((nodeId: string) => {
                 const index = nodes.findIndex(node => nodeId === node.uid || nodeId === nodeKey(node));
-                if (index > -1) {
+                const graphNode = graph.node(nodeId);
+                if (!graphNode.podGroup && index > -1) {
                     groupedNodeIds.push(nodeId);
+                } else {
+                    podGroupIds.push(nodeId);
                 }
-                graph.removeNode(nodeId);
-            });
-            graph.setNode(`${parentIds[0].toString()}/child/${kind}`, {
-                kind,
-                groupedNodeIds,
-                height: NODE_HEIGHT,
-                width: NODE_WIDTH,
-                count: nodeIds.length,
-                type: NODE_TYPES.groupedNodes
             });
-            graph.setEdge(parentIds[0].toString(), `${parentIds[0].toString()}/child/${kind}`);
+            const reducedNodeIds = nodeIds.reduce((acc, aNodeId) => {
```

Comment:
What is the use of `reducedNodeIds`?

Reply:
This is to keep track of the nodes that belong in the Group Node.  The original nodeIds contains the Pod Groups, but we need to pull them out.

Reply:
Look at the `count` where it is referenced.